main/src/library/MutList.flix

```diff
@@ -663,14 +663,9 @@ namespace MutList {
     /// Returns `v` as an array.
     ///
     pub def toArray(v: MutList[a]): Array[a] & Impure =
-        let l = length(v);
-        if (l == 0)
-            []
-        else {
-            let a = Array.new(get(0, v), l);
-            foreachWithIndex((x, i) -> a[l - i - 1] = x, v);
-            a
-        }
+        let MutList(a, _) = v;
+        shrink!(v);
+        deref a
```

Comment:
Does this not return the same arrays as the one inside MutList? Thus breaking its abstraction. We should always return a copy. I think `Copy.copy` can be used or slice it from beginning to end.

Reply:
Yes when length == capacity we return the same array. Oops.